### PR TITLE
Excerpts for Class Modifiers

### PR DESCRIPTION
**Pull request by khanhnwin**
Contributes to https://github.com/dart-lang/site-www/issues/4828

### CODE REVIEW

**Comment by parlough** [comment]
Everyone prepared some code excerpts! 😊

Thank you

**Comment by khanhnwin** [comment]
Cleared up the build - should be all green now, but definitely give the excerpts a review @parlough! Thank you!

File: src/language/class-modifiers.md [hunk 1]
```diff
@@ -47,27 +49,36 @@ entire interface, use the `abstract` modifier.
...
+base class Vehicle {
+  void moveForward(int meters) {
+    //...
+  }
 }
 ```
 
+<?code-excerpt "language/lib/class_modifiers/ex1/b.dart"?>
 ```dart
+
```
Comment (parlough) [inline src/language/class-modifiers.md:64]: Can we remove this extra line?

File: src/language/class-modifiers.md [hunk 2]
```diff
@@ -47,27 +49,36 @@ entire interface, use the `abstract` modifier.
...
 import 'a.dart';
 
-var myCar = Vehicle();       // Error: Cannot be constructed
+var myCar = Vehicle(); // Error: Cannot be constructed
 
-class Car extends Vehicle {  // Can be extended
-    int passengers;
-    // ...
+base class Car extends Vehicle {
+  // Can be extended
```
Comment (parlough) [inline src/language/class-modifiers.md:71]: Looking at this and the other "Can be" and "Cannot be" comments below, I'm a bit worried having them in the class declaration makes it look like it applies to the properties. What do you think about putting them above instead?

File: examples/language/lib/class_modifiers/ex3/a.dart [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Library a.dart
+interface class Vehicle {
+  void moveForward(int meters) {
+    //...
```
Comment (parlough) [inline examples/language/lib/class_modifiers/ex3/a.dart:4]: To be consistent with the other ellipsis comments:

```suggestion
    // ...
```

_ellipsis is always a hard word for me to spell for some reason??_

File: examples/language/lib/class_modifiers/ex3/b.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  // ERROR: Cannot be inherited
+  int passengers = 4;
+  // ...
+}
+
+class MockVehicle implements Vehicle {
+  // Can be implemented
+  @override
+  void moveForward(int meters) {
+    //...
```
Comment (parlough) [inline examples/language/lib/class_modifiers/ex3/b.dart:18]: To be consistent with the other ellipsis comments:

```suggestion
    // ...
```

File: examples/language/lib/class_modifiers/ex4/a.dart [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Library a.dart
+final class Vehicle {
+  void moveForward(int meters) {
+    //...
```
Comment (parlough) [inline examples/language/lib/class_modifiers/ex4/a.dart:4]: To be consistent with the other ellipsis comments:

```suggestion
    // ...
```

**Review by parlough — COMMENTED** [review]
Thanks **so much** for working on these @khanhnwin!

I left a few minor comments and one other for discussion.

File: examples/language/lib/class_modifiers/ex3/b.dart (hunk 4 above)
Comment (khanhnwin) [reply examples/language/lib/class_modifiers/ex3/b.dart:18]: applied!

File: examples/language/lib/class_modifiers/ex3/a.dart (hunk 3 above)
Comment (khanhnwin) [reply examples/language/lib/class_modifiers/ex3/a.dart:4]: haha!

Comment (khanhnwin) [reply examples/language/lib/class_modifiers/ex3/a.dart]: done!

File: src/language/class-modifiers.md (hunk 1 above)
Comment (khanhnwin) [reply src/language/class-modifiers.md]: Done!

**Comment by khanhnwin** [comment]
Alright, tweaked the excerpts based on your recommendations @parlough! Thanks for the review!!

**Comment by parlough** [comment]
Thanks for those adjustments! I just pushed some more fixes and cleanup that:

- Updates the `abstract` example to use `abstract` (looks like it was accidentally changed to use `base`)
- Move more error comments above the class declaration
- Standardizes some `// ...` comments

**Review by parlough — APPROVED** [review]
Okay, looks good to me! I pushed a few final fixes and changes if you want to take a look.

Thanks for adding excerpts to these. It helps a lot.

**Comment by khanhnwin** [comment]
Thank you @parlough!

**Pull request merged**